spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
+    def on_close(self, cancelable=False) -> None:
         """
         Perform actions before the plugin is closed.
 
-        This method **must** only operate on local attributes and not other
-        plugins.
+        .. caution::
+
+            This method **must** only operate on local attributes and not
+            call other plugins.
+
+        Parameters
+        ----------
+        cancelable: bool, optional
+            ``True`` if the close operation can potentially canceled;
+            ``False`` by default.
+
+        Returns
+        -------
+        None
         """
-        if hasattr(self, 'unregister'):
-            warnings.warn('The unregister method was deprecated and it '
-                          'was replaced by `on_close`. Please see the '
-                          'Spyder 5.2.0 migration guide to get more '
-                          'information.')
+        pass
 
     def can_close(self) -> bool:
         """
-        Determine if a plugin can be closed.
+        Determine if the plugin can be closed.
 
         Returns
         -------
         close: bool
-            True if the plugin can be closed, False otherwise.
+            ``True`` if the plugin can be closed, ``False`` otherwise.
         """
         return True
 
-    def update_font(self):
+    def update_font(self) -> None:
         """
-        This must be reimplemented by plugins that need to adjust their fonts.
+        Reimplemented by plugins that need to adjust their fonts.
 
         The following plugins illustrate the usage of this method:
-          * spyder/plugins/help/plugin.py
-          * spyder/plugins/onlinehelp/plugin.py
+          * :mod:`spyder.plugins.help.plugin`
+          * :mod:`spyder.plugins.onlinehelp.plugin`
+
+        Returns
+        -------
+        None
         """
         pass
 
-    def update_style(self):
+    def update_style(self) -> None:
         """
-        This must be reimplemented by plugins that need to adjust their style.
+        Reimplemented by plugins that need to adjust their style.
 
         Changing from the dark to the light interface theme might
         require specific styles or stylesheets to be applied. When
-        the theme is changed by the user through our Preferences,
+        the theme is changed by the user through our :guilabel:`Preferences`,
         this method will be called for all plugins.
+
+        Returns
+        -------
+        None
         """
         pass
 
-    def after_container_creation(self):
+    def after_container_creation(self) -> None:
         """
         Perform necessary operations before setting up the container.
 
-        This must be reimplemented by plugins whose containers emit signals in
-        on_option_update that need to be connected before applying those
-        options to our config system.
+        This must be reimplemented by plugins whose containers emit signals
+        that need to be connected.
```

Comment:
```suggestion
        This must be reimplemented by plugins whose containers emit signals
        that need to be connected before applying options to Spyder's config
        system.
```

The docstring doesn't make sense as it is.

Reply:
Oops, I forgot to apply your suggestion from Discord on this